kpops/cli/utils.py

```diff
@@ -15,10 +19,10 @@ def collect_pipeline_paths(pipeline_paths: Iterable[Path]) -> Iterator[Path]:
     :raises: ValueError: If `pipeline_path` is neither a file nor a directory.
     """
     for pipeline_path in pipeline_paths:
-        if pipeline_path.is_file():
+        if pipeline_path.is_file() and pipeline_path.name == PIPELINE_YAML:
```

Comment:
there should be a proper error if the entered name (e.g. `kpops generate pipelines/scratch/foo.yaml`) doesn't match the naming convention. We can skip the validation for now and implement it properly in a minor 6.x